File: files/CH-169.R
library(tidyverse)
library(readxl)

path = "files/CH-169 Extract From Text Part 2.xlsx"
input = read_excel(path, range = "B2:B7")
test  = read_excel(path, range = "D2:D7")

result = input %>%
  mutate(result = str_match_all(Text, "(?<=\\{)[^{}]+(?=\\})|(?<=\\[)[^\\]]+(?=\\])|(?<=\\()[^\\)]+(?=\\))|(?<=\\*)[^\\*]+(?=\\*)")) %>%
  mutate(result = map_chr(result, ~paste(., collapse = ", ")))

all.equal(result$result, test$Extracted, check.attributes = FALSE)
#> [1] TRUE

File: files/100-199/169/CH-169.py
import pandas as pd
import re

path = "CH-169 Extract From Text Part 2.xlsx"
input = pd.read_excel(path, usecols="B", skiprows=1, nrows=6)
test = pd.read_excel(path, usecols="D", skiprows=1, nrows=6)

def extract_text(text):
    pattern = r"(?<=\{)[^{}]+(?=\})|(?<=\[)[^\]]+(?=\])|(?<=\()[^\)]+(?=\))|(?<=\*)[^\*]+(?=\*)"
    matches = re.findall(pattern, text)
    return ", ".join(matches)

input['result'] = input['Text'].apply(extract_text)
 
print(input["result"].equals(test["Extracted"])) # True